cdc/scheduler/internal/v3/replication/replication_manager.go

```diff
@@ -499,9 +514,52 @@ func (r *Manager) AdvanceCheckpoint(
 	if slowestTableID != 0 {
 		r.slowestTableID = slowestTableID
 	}
+
+	// If changefeed's checkpoint lag is larger than 30s,
+	// log the 4 slowlest table infos every minute, which can
+	// help us find the problematic tables.
+	checkpointLag := currentTime.Sub(oracle.GetTimeFromTS(newCheckpointTs))
+	if checkpointLag > logSlowTablesLagThreshold &&
+		time.Since(r.lastLogSlowTablesTime) > logSlowTablesInterval {
+		r.logSlowTableInfo(currentTables, currentTime)
+		r.lastLogSlowTablesTime = time.Now()
+	}
+
 	return newCheckpointTs, newResolvedTs
 }
 
+func (r *Manager) logSlowTableInfo(currentTables []model.TableID, currentTime time.Time) {
+	// find the slow tables
+	for _, tableID := range currentTables {
+		table, ok := r.tables[tableID]
+		if !ok {
+			continue
+		}
+		heap.Push(&r.slowTableHeap, table)
```

Comment:
why not put  the condition " if lag < logSlowTablesLagThreshold " here?

Reply:
done